fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/AddDecodeNodeForDictStringRule.java

```diff
@@ -879,6 +879,13 @@ public Boolean visitBinaryPredicate(BinaryPredicateOperator predicate, Void cont
             if (!predicate.getChild(1).isConstant()) {
                 return false;
             }
+
+            Type leftType = predicate.getChild(0).getType();
+            // These type predicates couldn't push down to storage engine
+            if (leftType.isFloatingPointType() || leftType.isJsonType() || leftType.isTime()) {
```

Comment:
How about move these code to `Type.java` and give it a method name?

Reply:
These codes are too context-sensitive and not generate, So I think needn't abstract these codes to a method.